### PR TITLE
fix(python)!: Poetry python project is using deprecated `dev-dependencies` notation in  `pyproject.toml`

### PR DESCRIPTION
**Pull request by garysassano**
[Poetry 1.2.0](https://python-poetry.org/blog/announcing-poetry-1.2.0/) introduced the `group` notation for managing dependencies, which has been the recommended way to organize dependencies inside your `pyproject.toml` file for almost two years. The older `dev-dependencies` notation was deprecated, as detailed [here](https://python-poetry.org/docs/master/managing-dependencies/), and its use is discouraged.

BREAKING CHANGE: Poetry is now required to be at version 1.2.0 or above. Please update Poetry if needed.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/python/poetry.ts [hunk 1]
```diff
@@ -371,23 +371,30 @@ export class PoetryPyproject extends Component {
...
+          dependencies: dependencies,
+          group: {
+            dev: {
+              dependencies: devDependencies,
+            },
           },
         },
       },
+      "build-system": {
+        requires: ["poetry-core"],
```
Comment (mrgrain) [inline src/python/poetry.ts:390]: Given that we now require poetry >= 1.2.0 we should probably keep that in there.

**Review by mrgrain — CHANGES_REQUESTED** [review]
Can you also mark this as a breaking change with a note how to resolve it (ie. upgrade poetry). Thanks!

File: src/python/poetry.ts (hunk 1 above)
Comment (garysassano) [reply src/python/poetry.ts:390]: No, [poetry-core](https://github.com/python-poetry/poetry-core) is just the build system which is part of [poetry](https://github.com/python-poetry/poetry). It's a component which is kept separate so that other projects can make use of it.

What you want to do is currently not possible, despite being a feature that people have been asking for quite a long time as you can see from [this open issue](https://github.com/python-poetry/poetry/issues/3316).

Comment (garysassano) [reply src/python/poetry.ts:390]: It's a very common misconception, one that I also fell for in the past.

File: src/toml.ts [hunk 2]
```diff
@@ -22,10 +22,20 @@ export class TomlFile extends ObjectFile {
       return undefined;
     }
 
+    // Convert JSON content to TOML string
```
Comment (mrgrain) [inline src/toml.ts:25]: This seems entirely unrelated to this PR. Please keep it separate

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: src/toml.ts (hunk 2 above)
Comment (garysassano) [reply src/toml.ts:25]: How is it unrelated? Without modifying `toml.ts`, the formatting results in odd indentation that does not align with a fresh `pyproject.toml` file generated by `poetry init`. The aim is to produce a file through projen that appears as though it was directly created by poetry.

Comment (mrgrain) [reply src/toml.ts:25]: None of that is described in the PR or explicitly tested aside from the snapshot. I'd expect a bit of a write up about that and some investigation/reasoning why there is a discrepancy. 

Additionally I'm not at all convinced that that string manipulation is the right approach here. If the current TOML library is not capable of producing the desired output we will have to look for an alternative one. Swapping out the TOML package will definitely require a separate PR.

tl;dr All this just prevents a perfectly fine fix from being merged.

**Review by mrgrain — APPROVED** [review]
